components/UpdateProfileForm/steps/__tests__/MilitaryDetails.test.js

```diff
@@ -41,26 +40,24 @@ describe('UpdateProfileForm/Steps/MilitaryDetails', () => {
       </Formik>,
     );
 
-    const ReactSelect = wrapper.find('input').first();
+    const ReactSelect = container.querySelector('#react-select-branchOfService-input');
 
-    ReactSelect.simulate('blur')
-      .simulate('keyDown', { key: 'ArrowDown', keyCode: 40 })
-      .simulate('keyDown', { key: 'ArrowDown', keyCode: 40 });
+    fireEvent.blur(ReactSelect);
+    fireEvent.keyDown(ReactSelect, { key: 'ArrowDown', keyCode: 40 });
+    fireEvent.keyDown(ReactSelect, { key: 'ArrowDown', keyCode: 40 });
 
-    await asyncRenderDiff(wrapper);
+    fireEvent.keyDown(ReactSelect, { key: 'Enter', keyCode: 13 });
 
-    ReactSelect.simulate('keyDown', { key: 'Enter', keyCode: 13 });
+    fireEvent.change(container.querySelector('input#yearsOfService'), {
+      target: { id: 'yearsOfService', value: '3' },
+    });
 
-    await asyncRenderDiff(wrapper);
+    fireEvent.change(container.querySelector('input#payGrade'), {
+      target: { id: 'payGrade', value: 'E-5' },
+    });
 
-    wrapper
-      .find('input#yearsOfService')
-      .simulate('change', { target: { id: 'yearsOfService', value: '3' } });
-
-    wrapper.find('input#payGrade').simulate('change', { target: { id: 'payGrade', value: 'E-5' } });
-
-    wrapper.find('form').simulate('submit');
-    await asyncRenderDiff(wrapper);
+    fireEvent.submit(container.querySelector('form'));
+    await wait();
```

Comment:
```suggestion
    await wait(() => {
      expect(OperationCodeAPIMock.history.patch.length).toStrictEqual(1);
    });
```

Reply:
This suggestion applies to all the other files you touched I think too!

Reply:
got it.